### PR TITLE
add explorer template + test for explorer and block_creator

### LINKED ISSUE #2: template for explorer node

### LINKED ISSUE #6: Missing documenation and test
Need to complete the documentation for both templates (block_create, explorer).
Need explanation of the schema + example blueprint

And need unit tests

See example: https://github.com/zero-os/0-templates/tree/master/templates

### PR DESCRIPTION
**Pull request by zaibon**
closes #6 
closes #2

### CODE REVIEW

File: templates/block_creator/block_creator.py [hunk 1]
```diff
@@ -112,27 +125,26 @@ def start(self):
         start both tfchain daemon and client
         """
         self.state.check('actions', 'install', 'ok')
+        self.logger.info('Starting tfcaind %s', self.name)
```
Comment (FastGeert) [inline templates/block_creator/block_creator.py]: typo: tfchain

File: templates/block_creator/schema.capnp [hunk 2]
```diff
@@ -7,4 +7,5 @@ struct Schema {
     walletSeed @3: Text;
     walletPassphrase @4: Text;
     walletAddr @5: Text; # address of the wallet
+    network @6: Text="standard"; # network to join
```
Comment (FastGeert) [inline templates/block_creator/schema.capnp]: When we add a property to the schema with a default value, does it then get applied automatically to existing service schema's that are already running?

File: templates/explorer/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### Actions
+- `install`: create container with tfchain binaries.
+- `start`: starts the container and the tfchain daemon process and init wallet.
+- `stop`: stops the tfchain daemon process.
+- `consensus_stat`: return some statistics about the consensus
+- `gateway_stat`: return some statistics about the gateway
+
+### Examples:
+#### DLS:
```
Comment (FastGeert) [inline templates/explorer/README.md]: DSL (api interface):

File: templates/explorer/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+```python
+data = {'node':'node1', 'domain': 'explorer.tft.com'}
+explorer = robot.services.create('github.com/threefoldtoken/0-templates/explorer/0.0.1','explorer', data)
+explorer.schedule_action('install')
+explorer.schedule_action('start')
+```
+
+#### Blueprint:
```
Comment (FastGeert) [inline templates/explorer/README.md]: Blueprint (cli interface):

File: templates/block_creator/README.md [hunk 5]
```diff
@@ -11,11 +11,32 @@ This template is responsible for deploying block creator node.
...
 - `start`: starts the container and the tfchain daemon process and init wallet.
 - `stop`: stops the tfchain daemon process.
 - `wallet_address`: return wallet address
 - `wallet_amount`: return the amount of token in the wallet
-- `consensus_stat`: return some statistics about the consensus
\ No newline at end of file
+- `consensus_stat`: return some statistics about the consensus
+
+### Examples:
+#### DLS:
```
Comment (FastGeert) [inline templates/block_creator/README.md]: DSL (api interface):

File: templates/block_creator/README.md [hunk 6]
```diff
@@ -11,11 +11,32 @@ This template is responsible for deploying block creator node.
...
+```python
+data = {'node':'node1'}
+bc = robot.services.create('github.com/threefoldtoken/0-templates/block_creator/0.0.1','block_creator', data)
+bc.schedule_action('install')
+bc.schedule_action('start')
+```
+
+#### Blueprint:
```
Comment (FastGeert) [inline templates/block_creator/README.md]: Blueprint (cli interface):

File: templates/explorer/explorer.py [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    def _get_container(self):
+        sp = self._node_sal.storagepools.get('zos-cache')
+        try:
+            fs = sp.get(self.guid)
+        except ValueError:
+            fs = sp.create(self.guid)
+
+        # prepare persistant volume to mount into the container
+        node_fs = self._node_sal.client.filesystem
+        vol = os.path.join(fs.path, 'explorer')
```
Comment (FastGeert) [inline templates/explorer/explorer.py]: Trying to undesrtand why we use a fixed path 'explorer', and not something related to the running service. Will this not be blocking to run testnet and prodnet next to each other? Also, isn't this dangerous if we add other workloads beyond the tf chain on these boxes in the future?

File: templates/explorer/explorer.py [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+        self.state.delete('actions', 'install')
+        self.state.delete('wallet', 'init')
+
+    def start(self):
+        """
+        start both tfchain explorer and caddy
+        """
+        self.state.check('actions', 'install', 'ok')
+        self.logger.info('Starting tfcaind and caddy (%s)', self.name)
```
Comment (FastGeert) [inline templates/explorer/explorer.py]: typo: tfcaind

File: templates/block_creator/block_creator.py [hunk 9]
```diff
@@ -25,31 +25,41 @@ def __init__(self, name=None, guid=None, data=None):
         self.recurring_action('_monitor', 30)  # every 30 seconds
 
     @property
-    def node_sal(self):
+    def _node_sal(self):
         return j.clients.zero_os.sal.get_node(self.data['node'])
```
[the remark this answers is not in the document]
Comment (zaibon) [reply templates/block_creator/block_creator.py]: this logic belong in the client, you don't have to deal with that in this layer.
if you want implementation detail check : https://github.com/Jumpscale/lib9/blob/development/JumpScale9Lib/clients/zero_os/Client.py#L72

File: templates/block_creator/schema.capnp (hunk 2 above)
Comment (zaibon) [reply templates/block_creator/schema.capnp]: no

File: templates/explorer/explorer.py (hunk 7 above)
Comment (zaibon) [reply templates/explorer/explorer.py]: If you look line 55 I create a filesystem with the guid of the service, so there will be a different directory for every service. Using an extra explorer directory is to make it more easily readable in the mounts for debugging.

**Comment by zaibon** [comment]
Comments addressed

**Pull request merged**